File: Lab3P1/app/src/main/java/com/example/lab3p1/MainActivity.java
package com.example.lab3p1;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    EditText a, b;
    TextView c;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        a = (EditText) findViewById(R.id.currentName);
        b = (EditText) findViewById(R.id.currentName2);
        c = (TextView) findViewById(R.id.textView1);

    }


    public void Lugar1(View view){
        Intent lugar1 = new Intent (this, Main2Activity.class);
        startActivity(lugar1);
    }

    public void Lugar2(View view){
        Intent lugar1 = new Intent (this, Lugar2.class);
        startActivity(lugar1);
    }

    public void Lugar3(View view){
        Intent lugar1 = new Intent (this, Lugar3.class);
        startActivity(lugar1);
    }
    public void onClick(View view) {
        if (a.getVisibility() == View.VISIBLE){
            a.setVisibility(View.GONE);
            b.setVisibility(View.VISIBLE);
            c.setVisibility(View.GONE);

        }
        else
        {
            a.setVisibility(View.VISIBLE);
            b.setVisibility(View.GONE);
            c.setVisibility(View.VISIBLE);
        }
    }
}

File: Ejercicioenclase/ejercicioenclase/ejclase/ejclase/app/src/main/java/com/example/example/MainActivity.kt
package com.example.example

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.DataBinding
import android.view.View
import android.view.inputmethod.InputMethodManager


class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    val MyName: MyName = MyName("name1","nickname1")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        binding = DataBiningUtil.setContentView(this, R.layout.activity_main)
        binding.myName = myName

        binding.doneButton.setOnClickListener{
            addNickname(it)
        }
    }
    private fun addNickname(view: View){
        binding.apply{
            myName?.nickname = name_editor.text.toString()
            invalidateAll()
            name_edittext.visibility= View.GONE
            name_button.visibility = View.GONE
            name_edittext.visibiliry= View.VISIBLE
        }
        val im = getSystemSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        im.hideSoftInputFromWindow(view.windowToken, 0)
    }
}
